pkg/model/project.go

```diff
@@ -521,6 +521,12 @@ func (p *Project) DeleteRBACRole(name string) error {
 	return fmt.Errorf("%s role does nott exist", name)
 }
 
+// SetBuiltinRBAC sets built-in RBAC roles and user groups being assigned with it.
+func (p *Project) SetBuiltinRBAC() {
+	p.UserGroups = p.GetAllUserGroups()
+	p.RbacRoles = p.GetAllRBACRoles()
+}
+
```

Comment:
I feel a bit odd here, since if this one is only for set the builtin then it can be done at the time we make the project object (if I guess right), I mean I'm not a fan of setter, since if we have this method and forget to call it then project is not project hence it should be applied from the time we make the project object, wdyt?

Reply:
@khanhtc1202 

I feel your points and I was also thinking about it. 

How about doing it at the datastore layer like below?
https://github.com/pipe-cd/pipecd/blob/master/pkg/datastore/projectstore.go#L121-L127
```
func (s *projectStore) Get(ctx context.Context, id string) (*model.Project, error) {
	var entity model.Project
	if err := s.ds.Get(ctx, s.col, id, &entity); err != nil {
		return nil, err
	}
        entity.UserGroup = p.GetAllUserGroup()
        entity.RbacRoles = p.GetAllRBACRoles()
	return &entity, nil
}
```

By applying the above way, we need to not be conscious of whether a project contains built-in role or not besides, We can remove `GetAllProjectUserGroups` and  `GetAllProjectRBACRoles`.

Reply:
If it's for setting the builtin (pipecd defined rbac), then how about setting it on create project, and all other users defined one will update the project object by updateXXX interface?
I mean, can we set the project rbac builtin here
https://github.com/pipe-cd/pipecd/blob/master/pkg/app/ops/handler/handler.go#L195-L200

Reply:
There are several reasons for not storing built-in roles as a project in the DB.
One of them is able to update it simply when we add new resources.
So, I think we can choose from the below options.

① Fix `Get()` interface for projectStore by using `GetAllUserGroup()` and `GetAllRBACRoles()`.
```
func (s *projectStore) Get(ctx context.Context, id string) (*model.Project, error) {
	var entity model.Project
	if err := s.ds.Get(ctx, s.col, id, &entity); err != nil {
		return nil, err
	}
        entity.UserGroup = entity.GetAllUserGroup()
        entity.RbacRoles = entity.GetAllRBACRoles()
	return &entity, nil
}
```

② Fix `Get()` interface for projectStore by using `SetXXX` and delete `GetXXX`.
```
func (s *projectStore) Get(ctx context.Context, id string) (*model.Project, error) {
	var entity model.Project
	if err := s.ds.Get(ctx, s.col, id, &entity); err != nil {
		return nil, err
	}
        entity
                .SetBuiltinUserGroup()
                .SetBuiltinRBACRole()
	return &entity, nil
}
```

③ Handle this in the gateway layer in the same way as ①.
④ Handle this in the gateway layer in the same way as ②.

Reply:
thanks for your explanation, I remember it now 👍 then one vote from me for (2)

Reply:
Thank you for your vote and I also wanna vote ② because the caller just only needs to call `Get()`.

@nghialv 
wdyt?

Reply:
2 is also fine to me. 👍